cpp/src/decisiontree/decisiontree_impl.cuh

```diff
@@ -137,7 +136,7 @@ void build_treelite_tree(TreeBuilderHandle tree_builder,
           TREELITE_CHECK(TreeliteTreeBuilderSetLeafNode(
             tree_builder, q_node.unique_node_id, q_node.node.prediction));
         } else {
-          std::vector<double> leaf_vector(num_output_group);
+          std::vector<float> leaf_vector(num_output_group);
```

Comment:
Was this switch from double to single precision intentional? If so, should we test this somehow?

Reply:
Yes, since Treelite can only support float32 properly. I'd like to add support for float64 later.